src/test/test_main.cpp

```diff
@@ -0,0 +1,70 @@
+#include <random>
+
+#include "math.cpp"
+
+namespace Test {
+
+struct UnitTest {
+    const char *name;
+    bool flakey;
+
+    bool (* test)();
+};
+
+#define TEST(func) { STR(func), false, func }
+#define FLAKEY(func) { STR(func), true, func }
+
+void run_tests() {
+    UnitTest tests[] = {
+        TEST(math_typedef_s8),
+        TEST(math_typedef_u8),
+        TEST(math_typedef_s16),
+        TEST(math_typedef_u16),
+        TEST(math_typedef_s32),
+        TEST(math_typedef_u32),
+        TEST(math_typedef_f32),
+        TEST(math_typedef_s64),
+        TEST(math_typedef_u64),
+        TEST(math_typedef_f64),
+    };
+    int size = sizeof(tests) / sizeof(UnitTest);
+    int passed = 0;
+    int skipped = 0;
+    struct timespec start, end;
+
+    std::random_device rd;  // ensures random seed
+    std::mt19937 generator(rd());
+    std::shuffle(std::begin(tests), std::end(tests), generator);
+    printf("Running %d tests\n", size);
+
+    printf(HIDE_CURSOR);
```

Comment:
If you hide the cursor, you should maybe responde to the sigsegv, and some other signals with resetting the cursor. But meh, it's just something I thought of.